app.py

```diff
@@ -17,26 +18,93 @@ class User(UserMixin):
   def __init__(self,id):
     self.id = id
 
-items = [
-    {
-        'name': 'best buy_14DA0012DX_hp_chromebook',
-        'source': 'Best Buy',
-        'price': '$599.00',
-        'photo': 'https://pisces.bbystatic.com/image2/BestBuy_US/images/products/6365/6365772_sd.jpg;maxHeight=640;maxWidth=550',
-        'instock': 'Likely In Stock, Check Retailer',
-        'new': 'New',
-        'price_check': '2020-04-30 18:38:19.752171',
-    },
-    {
-        'name': 'best buy_14DA0012DX_hp_chromebook',
-        'source': 'Best Buy',
-        'price': '$599.00',
-        'photo': 'https://pisces.bbystatic.com/image2/BestBuy_US/images/products/6365/6365772_sd.jpg;maxHeight=640;maxWidth=550',
-        'instock': 'Likely In Stock, Check Retailer',
-        'new': 'New',
-        'price_check': '2020-04-30 18:38:19.752171',
-    },
-]
+
+#####################
+# PRODUCT INTERFACE #
+#####################
+def is_iterable(item):
+    try:
+        iter(item)
+        return True
+    except TypeError:
+        return False
+
+
+def list_from(item):
+    if not item:
+        return []
+    if isinstance(item, (str, dict)) or not is_iterable(
+        item
+    ):
+        return [item]
+    return list(item)
+
+
+class ProductDBInterface:
+    """Interface object so that the UI can easily obtain filtered products."""
+    def parse_sql_tuple(self, x):
+        return (ScrapedProduct(name = x[2], source = x[5], price = x[3], product_type=x[9], photo = x[4], instock= x[6], new = x[7], price_check= x[8]))
```

Comment:
I see that you added a 'type' column to the database, to put the changes into the db.py code it would just be the changes below:
_line 71_
`createRetailers_Products = "CREATE TABLE Retailers_Products (product_id int, retailer_id int, name varchar(64), price varchar(32), photo varchar(255), source varchar(255), instock varchar(32), new varchar(32), price_check varchar(64), type varchar(64), PRIMARY KEY (product_id, retailer_id), FOREIGN KEY (product_id) REFERENCES Products(product_id), FOREIGN KEY (retailer_id) REFERENCES Retailers(retailer_id))"`
_line 237_
`query = "REPLACE INTO Retailers_Products (product_id, retailer_id, name, price, photo, source, instock, new, price_check, type) VALUES (" + str(prodID) + "," + str(retID) + ",'" + str(x.name) + "','"+ str(x.price) +"','"+str(x.photo)+"','"+str(x.source)+"','"+str(x.instock)+"','"+str(x.new)+"','"+str(x.price_check)+"','" + str(x.product_type) + "');"`